cluster-autoscaler/utils/drain/drain.go

```diff
@@ -231,6 +158,104 @@ func GetPodsForDeletionOnNodeDrain(
 	return pods, daemonSetPods, nil, nil
 }
 
+func legacyCheckForReplicatedPods(listers kube_util.ListerRegistry, pod *apiv1.Pod, minReplica int32) (replicated bool, isDaemonSetPod bool, blockingPod *BlockingPod, err error) {
+	replicated = false
+	refKind := ""
+	checkReferences := listers != nil
+	isDaemonSetPod = false
+
+	controllerRef := ControllerRef(pod)
```

Comment:
This is the same code as our current code [here](https://github.com/kubernetes/autoscaler/pull/5507/files#diff-a22fbdb27ac8bbd8f9c9597042678aa5f6541d207160adc775d56e13f3fe99beL115-L201) abstracted into a function. I have done some minor changes like:
1. Change return values to include `replicated` and `isDaemonSetPod`. Also, remove `[]*apiv1.Pod` since we don't need to return it from this function. Returning pod list is done in the `else` part [here](https://github.com/kubernetes/autoscaler/pull/5507/files#diff-a22fbdb27ac8bbd8f9c9597042678aa5f6541d207160adc775d56e13f3fe99beR126-R127). 
2. Don't do appending to `daemonSetPods` slice here. Instead just return if `isDaemonSetPod` and [let the calling function do the appending](https://github.com/kubernetes/autoscaler/pull/5507/files#diff-a22fbdb27ac8bbd8f9c9597042678aa5f6541d207160adc775d56e13f3fe99beR131-R133).